ui/components/SourceDetail.tsx

```diff
@@ -57,6 +57,10 @@ function SourceDetail({ className, name, info, type }: Props) {
       return true;
     }
 
+    if (a?.helmChart?.sourceRef?.kind == s.type && a?.helmChart?.name == name){
+      return true;
+    }
+
```

Comment:
No objection on this, just a note, we can simplify this a bit with a function definition outside of the filter loop:
```typescript
const isRelevant(expectedType, expectedName) {
  return (expectedType == s.type && expectedName == name)
}
```

and later just use

```typescript
  if (isRelevant(a?.sourceRef?.kind, a?.sourceRef?.name)) { return true }
  if (isRelevant(a?.helmChart?.sourceRef?.kind, a?.helmChart?.name)) { return true }
  
  return false;
```

or even chaining them and

```typescript
return isRelevant(a?.sourceRef?.kind, a?.sourceRef?.name) ||
       isRelevant(a?.helmChart?.sourceRef?.kind, a?.helmChart?.name)
```